### PR TITLE
[Feature:InstructorUI] Add registration subsection

### PR DESCRIPTION
**Pull request by vikrsri**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
There is currently no way to put students into subsections, allowing further sorting of students.
Addresses #6064.

### What is the new behavior?
This PR adds a registration subsection that can be set by an instructor. Further PR's will allow sorting and grading by students' registration subsections.

The subsection can be set when an instructor either edits or adds a new student in the Manage Students page. If a subsection has been set for a student, it will be appended to the student's registration section in the Manage Students page. Otherwise, there will be no change.

### CODE REVIEW

File: migration/migrator/migrations/course/20210113174909_registration_subsection.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+def up(config, database, semester, course):
+    database.execute('ALTER TABLE users ADD COLUMN registration_subsection character varying(255)')
+
+
+def down(config, database, semester, course):
+    database.execute('ALTER TABLE users DROP IF EXISTS registration_subsection')
```
Comment (bmcutler) [inline migration/migrator/migrations/course/20210113174909_registration_subsection.py:7]: I don't think it should be necessary to delete this column in the down migration (we don't want to lose data if we temporarily revert this PR).

File: migration/migrator/migrations/course/20210113174909_registration_subsection.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+def up(config, database, semester, course):
+    database.execute('ALTER TABLE users ADD COLUMN registration_subsection character varying(255)')
```
Comment (bmcutler) [inline migration/migrator/migrations/course/20210113174909_registration_subsection.py:3]: Make this an 'add if does not exist' 
(see other migration files for exact syntax)

**Review by bmcutler — CHANGES_REQUESTED** [review]

**Review by bmcutler — APPROVED** [review]
Quick code review and quick test look ok!  Thanks!
This is what we agreed for the step 1 PR.

@MasterOdin -- @vikrsri and I discussed the unexpected additional edits to the .sql file (changes were made using the postgres dump instructions... ).  Can you please review and advise?

File: site/app/models/User.php [hunk 3]
```diff
@@ -388,6 +394,10 @@ public function updateUserNotificationSettings($key, $value) {
         $this->notification_settings[$key] = $value;
     }
 
+    public function getRegistrationSubsection() {
+        return $this->registration_subsection;
+    }
```
Comment (MasterOdin) [inline site/app/models/User.php:399]: you can get rid of this function, it's not needed (as using `@prop` annotation automatically makes available this getter field.

File: site/app/models/User.php [hunk 4]
```diff
@@ -418,6 +428,10 @@ public function setRegistrationSection($section) {
         $this->registration_section = ($section !== null) ? $section : null;
     }
 
+    public function setRegistrationSubsection($section) {
+        $this->registration_subsection = $section;
+    }
+
```
Comment (MasterOdin) [inline site/app/models/User.php:434]: you can get rid of this function, it's not needed (as using `@prop` annotation automatically makes available this setter field.

**Review by MasterOdin — APPROVED** [review]

**Comment by MasterOdin** [comment]
The SQL file changes are fine. The `gradeable_access` table was added by hand in #5623 and then no one has run the dump DB command since then. The reason for that it looks like `gradeable_access`'s schema changed is that `SERIAL` is a pseudo-type in postgresql that translates under the hood to creating a sequence, assigning ownership, etc. It's a shorthand that the pg_dump command does not use, as it is more inline with how the data structures are actually stored in the DB.

**Pull request merged**